pkg/cmd/prebuild/update.go

```diff
@@ -43,19 +45,56 @@ var prebuildUpdateCmd = &cobra.Command{
 			return nil
 		}
 
-		if len(args) < 2 {
+		// Determine the mode of operation: interactive or non-interactive
+		if len(args) == 2 || (branchFlag != "" || retentionFlag != 0 || commitIntervalFlag != 0 || len(triggerFilesFlag) > 0) {
+			// Non-interactive mode: use provided arguments and flags
+			if len(args) < 2 {
+				return errors.New("Both project config name and prebuild ID must be specified when using flags")
+			}
+
+			projectConfigRecieved = args[0]
+			prebuildID := args[1]
+
+			prebuild, res, err = apiClient.PrebuildAPI.GetPrebuild(ctx, projectConfigRecieved, prebuildID).Execute()
+			if err != nil {
+				return apiclient_util.HandleErrorResponse(res, err)
+			}
+
+			// Set prebuild details based on flags
+			if branchFlag != "" {
+				prebuild.Branch = branchFlag
+			}
+
+			if retentionFlag > 0 {
+				prebuild.Retention = int32(retentionFlag)
+			}
+
+			if commitIntervalFlag > 0 {
+				prebuild.CommitInterval = util.Pointer(int32(commitIntervalFlag))
+			}
+
+			if len(triggerFilesFlag) > 0 {
+				prebuild.TriggerFiles = triggerFilesFlag
+			}
+			prebuildAddView.Branch = prebuild.Branch
+			prebuildAddView.Retention = strconv.Itoa(int(prebuild.Retention))
+			prebuildAddView.ProjectConfigName = projectConfigRecieved
+			prebuildAddView.TriggerFiles = triggerFilesFlag
+			prebuildAddView.CommitInterval = strconv.Itoa(int(commitIntervalFlag))
```

Comment:
I think the right sides in these assignments should be from the `prebuild` object instead of the flags

Reply:
modified it